### ISSUE TITLE
issues.list() not working

### ISSUE THREAD

**Issue by joecarney** [michael/github#292]
```
    issues.list({}, function (err, issues) {
```

is timing out when I use it.  

---

The reason is this... Inside `src/github.js`, your `iterate() _request` function.. 

```
      var _requestAllPages = Github._requestAllPages = function _requestAllPages(path, cb) {
         var results = [];

         (function iterate() {
            _request('GET', path, null, function (err, res, xhr) {

```

is returning a **xhr** value as **undefined** for me...

As a very quick/dirty test, just to to see if I could get some results back, i added a `if (!hxr) {cb}` and it started to work fine...

```
               if (!(res instanceof Array)) {
                  res = [res];
               }

               results.push.apply(results, res);

               if (!xhr) {
                  cb(err, results);
               }
```

You may want to take a look into this :)

**Comment by AurelioDeRosa** [comment]
The `list()` function is supposed to iterate because you might have more than 100 issues, so we implemented the correct behavior for that feature. Also, we [have a test to check if XHR is not undefined](https://github.com/michael/github/blob/3dc854a40e3cca355ee1258fe4956ff23c7800fc/test/test.issue.js#L35).

Can you create a JSFiddle to demonstrate your problem? How are you including the library?

**Comment by joecarney** [comment]
At the moment, all I've done is written 2 tests in Jasmine, one to connect to the repo and one to load the issues.  Connection works fine, loading the issues timeouts.  That is why I went into your code to see what is happening

```
var Github = require('github-api');

var github = new Github({
  token: "xxx",
  auth: "oauth"
});

var _username = "joecarney";
var _reponame = "xxx";
var _repo = github.getRepo(_username, _reponame);
var _issues = github.getIssues(_username, _reponame);

describe("Checking github connection", function () {

  it("the repo has the correct name", function (done) {

    _repo.show(function (err, repo) {
      //console.log('err');
      //console.log(err);
      //console.log('repo');
      //console.log(repo);

      expect(repo.name)
        .toBe("xxx");

      done();
    });

  });

  it("the issues load", function (done) {

    _issues.list({}, function (err, issues) {
      console.log('err');
      console.log(err);

      console.log('issues');
      console.log(issues);

      expect(err)
        .toBe(null);

      done();
    });

  });

});
```

This is all I've written so far, was just testing the library (with real tokens/repo names) to see if I could work with it.

**Comment by joecarney** [comment]
Debugging your code to see what is happening...

```
      var _requestAllPages = Github._requestAllPages = function _requestAllPages(path, cb) {
         var results = [];

         (function iterate() {
            _request('GET', path, null, function (err, res, xhr) {
               console.log('err');
               console.log(err);
               console.log('res');
               console.log(res);
               console.log('xhr');
               console.log(xhr);

               if (err) {
                  console.log('calling back an err');
                  return cb(err);
               }

               if (!(res instanceof Array)) {
                  console.log('creating an instance of the array');
                  res = [res];
               }

               console.log('set the results');
               results.push.apply(results, res);

               var links = (xhr.getResponseHeader('link') || '')
                  .split(/\s*,\s*/g);
               console.log('links');
               console.log(links);

               var next = false;

               links.forEach(function (link) {
                  next = /rel="next"/.test(link) ? link : next;
                  console.log('next');
                  console.log(next);
               });
```

Gives an output of...

```
err
null
res
[ { url: 'https://api.github.com/repos/joecarney'...(res of the res object displayed here) } ]
xhr
undefined             <-- This is what is causing the timeout in your code 
set the results
```

**Comment by angelxmoreno** [comment]
I think I am having a similiar issue with `user.orgRepos` that also makes use of `_requestAllPages`. I am getting a `TypeError: Cannot read property 'getResponseHeader' of undefined` when running into this code:
```js
 var links = (xhr.getResponseHeader('link') || '').split(/\s*,\s*/g);
```

To get the error I changed that line to 
```js
try {
    var links = (xhr.getResponseHeader('link') || '').split(/\s*,\s*/g);
} catch (e) {
    cb(e);
}

**Comment by AurelioDeRosa** [comment]
Can you provide us a JSfiddle that demonstrate the problem? I'm not able to replicate the issue and the tests cover this function.

**Comment by YUGI-YUGI** [comment]
@AurelioDeRosa  im also facing same issue .

`
        var github = new Github();
	var issues = github.getIssues("Shippable", "support");
	issues.list( {page:4},function(err, issues) {
		console.log(issues);
	});
`
and im facing problem like in console.
![screen shot 2016-04-12 at 2 04 53 pm](https://cloud.githubusercontent.com/assets/13832271/14454331/9452b91a-00b7-11e6-9767-4f3964681d3a.png)


> github.js:128 Uncaught (in promise) TypeError: Cannot read property 'getResponseHeader' of undefined(…)(anonymous function) @ github.js:128(anonymous function) @ github.js:91

**Comment by redism** [comment]
I had exactly the same problem. (node v5.7.0) and here is what I found.

![axios](https://cloud.githubusercontent.com/assets/510218/14759628/268c0836-0964-11e6-86fb-cd9da927531c.png)

I don't know if axios interface has been changed, but my installed axios version is 0.9.0
Passing response.headers and parsing link from headers solves the problem, but I'm not sure it's proper solution.

**Comment by clayreimann** [comment]
Closed by e0666fd
